pkg/config/config.go

```diff
@@ -245,6 +248,27 @@ func GetMockACPIPowerPath() string {
 	return MockACPIPowerPath
 }
 
+// GetMachineSpec initializes map of MachineSpecValues from MACHINE_SPEC
+func GetMachineSpec() *MachineSpec {
```

Comment:
Could we write a unit-test for this? 
Could you also please consider refactoring this to avoid deeply nested paths? 


```
// GetMachineSpec initializes a map of MachineSpecValues from MACHINE_SPEC
func GetMachineSpec() *MachineSpec {
	// Retrieve the machine ID from configuration
	machineID := getConfig(MachineIDKey, "")
	if machineID == "" {
		// If machine ID is not found, generate a default machine spec
		return generateSpec()
	}

	specPath := filepath.Join(MachineSpecPath, machineID)
	spec, err := readMachineSpec(specPath)
	if err  != nil {
 		klog.Errorf("failed to read spec file: %s: %v; using default spec", specPath, err);
 		return generateSpec()
	}
 	return spec
}

func readMachineSpec(path string) (MachineSpec, error) {
	...
}
```